### PR TITLE
Updated Unit tests to use latest Unit Test templates, Added new Unit tests - fixes #27

### LINKED ISSUE #27: Unit tests need to be updated to use Standard Test Template
I will submit a PR.

### PR DESCRIPTION
**Pull request by PlagueHO**
* Unit tests updated to use standard unit test templates.
* MSFT_xDnsServerZoneTransfer: Added unit tests.
                               Updated to meet Style Guidelines.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xdnsserver/29)

### CODE REVIEW

**Comment by TravisEz13** [comment]
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdnsserver/29)*

**Comment by TravisEz13** [comment]
I'll allow some time for others to review.  @iainbrighton ?

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-:-KEmlwd572ZwLOq3Rbwv:1696911257)*

File: DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1 [hunk 1]
```diff
@@ -40,7 +40,10 @@ function Get-TargetResource
 
 #endregion
 
-    $currentZone = Get-CimInstance -ClassName MicrosoftDNS_Zone -Namespace root\MicrosoftDNS -Verbose:$false | ?{$_.Name -eq $Name}
+    $currentZone = Get-CimInstance `
+        -ClassName MicrosoftDNS_Zone `
+        -Namespace root\MicrosoftDNS `
+        -Verbose:$false | ?{$_.Name -eq $Name}
```
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1]: Shouldn't this be:
```powershell
-Verbose:$false | Where-Object { $_.Name -eq $Name }
```

File: DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1 [hunk 2]
```diff
@@ -117,71 +126,100 @@ function Validate-ResourceProperties
...
+    $checkZoneMessage = $($LocalizedData.CheckingZoneMessage) `
+        -f $Name
     Write-Verbose -Message $checkZoneMessage
  
     # Get the current value of transfer zone
-    $currentZone = Get-CimInstance -ClassName MicrosoftDNS_Zone -Namespace root\MicrosoftDNS -Verbose:$false | ? {$_.Name -eq $Name}
+    $currentZone = Get-CimInstance `
+        -ClassName MicrosoftDNS_Zone `
+        -Namespace root\MicrosoftDNS `
+        -Verbose:$false | ? {$_.Name -eq $Name}
```
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1]: Same again - should be:
```powershell
-Verbose:$false | Where-Object { $_.Name -eq $Name }
```

**Comment by TravisEz13** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1, line 43 \[r1\]](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-KErX3tbsGG9xQrMVBs_:-KErX3tc81S5gZnCZPpk:339252471) ([raw file](https://github.com/powershell/xdnsserver/blob/517f7cf3e012d4067fa03ff5e4588dbe5b2bef9f/DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1#L43)):*
not specific to this line: 
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

---

*[DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1, line 46 \[r1\]](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-KEhl6577N-nGwWSg6Qq-r1-46:-KErWbG2IpNn7r5_Gi-o:-2062585774) ([raw file](https://github.com/powershell/xdnsserver/blob/517f7cf3e012d4067fa03ff5e4588dbe5b2bef9f/DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1#L46)):*
yes, but I believe this is just a warning in the script analyzer.

---

*[DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1, line 137 \[r1\]](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-KEhl6577N-nGwWSg6Qq-r1-137:-KErWimjm7zevNUXaLjf:-456158593) ([raw file](https://github.com/powershell/xdnsserver/blob/517f7cf3e012d4067fa03ff5e4588dbe5b2bef9f/DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1#L137)):*
same

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdnsserver/29)*

**Comment by PlagueHO** [comment]
Review status: 5 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1, line 43 \[r1\]](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-KErX3tbsGG9xQrMVBs_:-KErsFVIy2m6WhTsC2gH:1903026517) ([raw file](https://github.com/powershell/xdnsserver/blob/517f7cf3e012d4067fa03ff5e4588dbe5b2bef9f/DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1#L43)):*
Doh, I expected that - had two PR's waiting and wasn't sure which branch would be accepted first. Fixed.

---

*[DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1, line 46 \[r1\]](https://reviewable.io:443/reviews/powershell/xdnsserver/29#-KEhl6577N-nGwWSg6Qq-r1-46:-KErrpgVAFECPfyDEww7:1705729012) ([raw file](https://github.com/powershell/xdnsserver/blob/517f7cf3e012d4067fa03ff5e4588dbe5b2bef9f/DSCResources/MSFT_xDnsServerZoneTransfer/MSFT_xDnsServerZoneTransfer.psm1#L46)):*
Good catch, best to fix it now as there is an issue being discussed to enable PSSA Warning level validation.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdnsserver/29)*

**Comment by TravisEz13** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdnsserver/29)*

**Pull request merged**